### PR TITLE
Wip/gportay/genesys changes

### PR DESCRIPTION
**Pull request by gportay**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/genesys/fu-genesys-scaler-device.c [hunk 1]
```diff
@@ -603,14 +603,14 @@ fu_genesys_scaler_device_pause_r2_cpu(FuGenesysScalerDevice *self, GError **erro
 		return FALSE;
 	}
 
-	g_usleep(200000); /* 200ms */
+	g_usleep(200000); /* 200 ms */
```
[the remark this answers is not in the document]
Comment (gportay) [reply plugins/genesys/fu-genesys-scaler-device.c:606]: Ah, okay, I did for the steelseries.

I will revert.

File: plugins/genesys/fu-genesys-scaler-firmware.c [hunk 2]
```diff
@@ -70,14 +72,26 @@ fu_genesys_scaler_firmware_parse(FuFirmware *firmware,
...
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_FILE,
+				    "invalid address, expected a multiple of 0x%x, and got 0x%x",
+				    GENESYS_SCALER_BANK_SIZE,
+				    addr);
+			return FALSE;
+		}
+		self->second_image_program_addr = addr;
+		fu_firmware_set_addr(firmware, addr);
```
[the remark this answers is not in the document]
Comment (gportay) [reply plugins/genesys/fu-genesys-scaler-firmware.c:94]: Okay, I will have a look at it.

[the remark this answers is not in the document]
Comment (gportay) [reply plugins/genesys/fu-genesys-scaler-firmware.c:94]: Hum, I am not sure how to use it.

[the remark this answers is not in the document]
Comment (gportay) [reply plugins/genesys/fu-genesys-scaler-firmware.c:94]: However, I have a commit that will remove all the Footer "bloat" code.

The footer was originally copied from the original crypt-ed firmware; then I had to tweak it.

Finally, I will drop it and provide the decrypted firmware + the public-key append which is used to ensure we write the right blob to the right device.

[the remark this answers is not in the document]
Comment (gportay) [reply plugins/genesys/fu-genesys-scaler-firmware.c:94]: What do you think?

Do you want I add the commit to that PR? or do it later when I ensure it is working stable?

**Comment by gportay** [comment]
@hughsie: How can I fix https://github.com/fwupd/fwupd/runs/6721862721?check_suite_focus=true?

File: plugins/genesys/README.md [hunk 3]
```diff
@@ -126,6 +126,15 @@ Scaler GPIO value.
 
 Since 1.7.6.
 
+### GenesysScalerCfiFlashId
+
+Cfi Flash Id.
```
Comment (hughsie) [inline plugins/genesys/README.md:131]: CFi pls

Comment (hughsie) [reply plugins/genesys/README.md:131]: Gah, CFI

**Comment by hughsie** [comment]
Maybe squash them down a bit too?

File: plugins/genesys/README.md (hunk 3 above)
Comment (gportay) [reply plugins/genesys/README.md:131]: ok

**Comment by gportay** [comment]
> Maybe squash them down a bit too?

I can, but they are quite independent :/

**Comment by gportay** [comment]
I can probably squash few more to the first commit.

File: plugins/genesys/fu-genesys-scaler-firmware.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (hughsie) [reply plugins/genesys/fu-genesys-scaler-firmware.c:92]: Up to you dude, your call.

**Review by hughsie — APPROVED** [review]

**Pull request merged**